app/Http/Livewire/Like.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace App\Http\Livewire;
+
+use App\User;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Facades\Auth;
+use Livewire\Component;
+
+class Like extends Component
+{
+    public $likes;
+
+    public $likeableId;
+
+    public $likeableType;
+
+    /**
+     * Mount the livewire component.
+     *
+     * @param \Illuminate\Database\Eloquent\Model $likeable
+     * @return void
+     */
+    public function mount($likeable): void
+    {
+        $this->likeableId = $likeable->id;
+        $this->likeableType = get_class($likeable);
+        $this->likes = $this->likeable->likes_count;
+    }
+
+    /**
+     * Get the likeable property. This will be a thread or reply eloquent model.
+     *
+     * @return Illuminate\Database\Eloquent\Model
+     * @throws ModelNotFoundException
+     */
+    public function getLikeableProperty(): Model
+    {
+        return $this->likeableType::findOrFail($this->likeableId);
+    }
+
+    /**
+     * Get the user of the current session.
+     *
+     * @return \App\Models\User
+     */
+    public function getUserProperty(): ?User
+    {
+        return Auth::user();
+    }
+
+    /**
+     * Toggle the like count for the logged in user.
+     *
+     * @return void
+     */
+    public function toggleLike(): void
+    {
+        if (! $this->user) {
+            return;
+        }
+
+        $this->likeable->isLikedBy($this->user) ? $this->likeable->dislikedBy($this->user) : $this->likeable->likedBy($this->user);
```

Comment:
Use an if/else statement for this long one-liner so the code becomes a little more readable.